### PR TITLE
Markus/vir 1744 set job field on new samples

### PR DESCRIPTION
**Pull request by SnickerDoodlePop**

### CODE REVIEW

**Review by ReeceHoffmann — CHANGES_REQUESTED** [review]
Looks like it solves the atomicity problem! Need to tweak how `create_job` is getting access to the data sources it needs though. Looking close

File: virtool/jobs/data.py [hunk 1]
```diff
@@ -201,27 +202,20 @@ async def create(
...
-
-        if job_id:
-            document["_id"] = job_id
-
-        document = await self._db.jobs.insert_one(document)
-        await self._client.enqueue(workflow, document["_id"])
 
-        return await fetch_complete_job(self._db, document)
+        job_mongo = copy.copy(self._db)
+        job_client = copy.copy(self._client)
```
Comment (igboyes) [inline virtool/jobs/data.py:207]: Why?

**Review by igboyes — COMMENTED** [review]
Looking good other than one oddity.

File: virtool/jobs/data.py (hunk 1 above)
Comment (SnickerDoodlePop) [reply virtool/jobs/data.py:207]: oops, that shouldn't be there

**Pull request merged**